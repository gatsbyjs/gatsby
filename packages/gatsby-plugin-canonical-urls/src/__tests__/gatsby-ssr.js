const { onRenderBody } = require(`../gatsby-ssr`)

describe(`gatsby-plugin-canonical-urls`, () => {
  it(`creates a canonical link if siteUrl is set`, async () => {
    const pluginOptions = {
      siteUrl: `http://someurl.com`,
    }
    const setHeadComponents = jest.fn()
    const pathname = `/somepost`

    await onRenderBody(
      {
        setHeadComponents,
        pathname,
      },
      pluginOptions
    )

    expect(setHeadComponents).toMatchSnapshot()
    expect(setHeadComponents).toHaveBeenCalledTimes(1)
  })

  it(`strips a trailing slash on the siteUrl and leaves search parameters`, async () => {
    const pluginOptions = {
      siteUrl: `http://someurl.com/`,
    }
    const setHeadComponents = jest.fn()
    const pathname = `/somepost?tag=foobar`

    await onRenderBody(
      {
        setHeadComponents,
        pathname,
      },
      pluginOptions
    )

    expect(setHeadComponents).toMatchSnapshot()
    expect(setHeadComponents).toHaveBeenCalledTimes(1)
  })

  it(`strips search parameters if option stripQueryString is true`, async () => {
    const pluginOptions = {
      siteUrl: `http://someurl.com`,
      stripQueryString: true,
    }
    const setHeadComponents = jest.fn()
    const pathname = `/somepost?tag=foobar`

    await onRenderBody(
      {
        setHeadComponents,
        pathname,
      },
      pluginOptions
    )

    expect(setHeadComponents).toMatchSnapshot()
    expect(setHeadComponents).toHaveBeenCalledTimes(1)
  })

  it(`does not create a canonical link if siteUrl is not set`, async () => {
    const pluginOptions = {}
    const setHeadComponents = jest.fn()
    const pathname = `/somepost`

    await onRenderBody(
      {
        setHeadComponents,
        pathname,
      },
      pluginOptions
    )

    expect(setHeadComponents).toMatchSnapshot()
    expect(setHeadComponents).toHaveBeenCalledTimes(0)
  })

  it(`does not create a canonical link if siteUrl is set but path is excluded`, async () => {
    const pluginOptions = {
      siteUrl: `http://someurl.com`,
      excludes: [`/somepost`, `/excluded-page-two`],
    }
    const setHeadComponents = jest.fn()
    const pathname = `/somepost`

    await onRenderBody(
      {
        setHeadComponents,
        pathname,
      },
      pluginOptions
    )

    expect(setHeadComponents).toMatchSnapshot()
    expect(setHeadComponents).toHaveBeenCalledTimes(0)
  })
})
