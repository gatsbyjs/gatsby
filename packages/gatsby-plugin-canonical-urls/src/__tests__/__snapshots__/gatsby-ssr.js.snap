// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-canonical-urls creates a canonical link if siteUrl is set 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        <link
          data-basehost="someurl.com"
          data-baseprotocol="http:"
          href="http://someurl.com/somepost"
          rel="canonical"
        />,
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-canonical-urls does not create a canonical link if siteUrl is not set 1`] = `[MockFunction]`;

exports[`gatsby-plugin-canonical-urls strips a trailing slash on the siteUrl and leaves search parameters 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        <link
          data-basehost="someurl.com"
          data-baseprotocol="http:"
          href="http://someurl.com/somepost?tag=foobar"
          rel="canonical"
        />,
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-canonical-urls strips search parameters if option stripQueryString is true 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        <link
          data-basehost="someurl.com"
          data-baseprotocol="http:"
          href="http://someurl.com/somepost"
          rel="canonical"
        />,
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-canonical-urls does not create a canonical link if siteUrl is set but path is excluded 1`] = `[MockFunction]`;
